src/main/java/apaintus/controllers/CanvasController.java

```diff
@@ -56,71 +60,62 @@ public void injectParentController(Controller controller) {
         this.updateService = new UpdateService(this.attributeController, this.toolBarController);
 
         snapGrid = new SnapGrid(toolBarController.getGridSpacing(),canvas.getWidth(),canvas.getHeight(),toolBarController.getStrokeSize(),false);
+        invoker = this.controller.getInvoker();
     }
 
     @Override
     public void initialize() {
+
         canvas.setOnMousePressed(event -> {
             lastMouseClickPosition = new Point(event.getX(), event.getY());
-
-            if (activeShape != null) {
-                activeShape.setSelected(false);
-                this.attributeController.resetAttributes();
-                redrawCanvas();
-            }
-
             activeTool = toolBarController.getActiveTool();
-            if (activeTool == ActiveTool.SELECT) {
-                ListIterator<DrawableShape> iterator = drawnShapes.listIterator(drawnShapes.size());
-                while(iterator.hasPrevious()) {
-                	DrawableShape shape = (DrawableShape)iterator.previous();
-                    if (shape.contains(lastMouseClickPosition)) {
-                        activeShape.setSelected(false);
-
-                        activeShape = shape;
-                        attributeController.update(activeShape);
-                        toolBarController.update(activeShape.getShapeAttributes());
-                        activeShape.setSelected(true);
-                        redrawCanvas();
-                        return; // we found the first shape, no need to
-                    }
-                }
-            }
-            activeShape = canvasService.createShape(activeTool, event, getCanvasDimension(), snapGrid);
+
+            tempActiveShape = canvasService.createShape(activeTool, event, getCanvasDimension(), snapGrid);
+            deselectCommand = new DeselectCommand(this, attributeController, activeShape);
+            deselectCommand.execute();
 
             if (activeTool == ActiveTool.SELECT) {
-                selectionBox = (SelectionBox) activeShape;
+                selectionBox = (SelectionBox) tempActiveShape;
             }
         });
 
         canvas.setOnMouseReleased(event -> {
+            deselectCommand.undo();
+
             if (event.getX() == lastMouseClickPosition.getX() && event.getY() == lastMouseClickPosition.getY()) {
-                return;
-            }
+                if (activeTool == ActiveTool.SELECT) {
```

Comment:
why do we do something when we click and we don't move. I don't get the logic behind. previously we were not doing anything, why the change then ?

Reply:
Quand tu click et t'as pas bouge, il check pour une selection / deselection